### PR TITLE
Misc additions and changes to CI documentation generation

### PR DESCRIPTION
**Pull request by fredmorcos**
### Short description
- Add short commit hash to html doc artifacts that are uploaded to GH.
- Always build and GH artifact upload documentation PDFs since installing their deps doesn't take that much time anymore.
- Add the product (auth/rec/dnsdist) name to the html doc artifacts.

### Checklist
I have:
- [X] read the [CONTRIBUTING.md](https://github.com/PowerDNS/pdns/blob/master/CONTRIBUTING.md) document
- [ ] compiled this code
- [ ] tested this code
- [ ] included documentation (including possible behaviour changes)
- [ ] documented the code
- [ ] added or modified regression test(s)
- [ ] added or modified unit test(s)

### CODE REVIEW

**Comment by Habbie** [comment]
can you add the hash to the pdf names too?

**Comment by fredmorcos** [comment]
> can you add the hash to the pdf names too?

Done.

**Comment by fredmorcos** [comment]
Can we merge this?

**Review by rgacogne — COMMENTED** [review]
So the changes look good to me, but I feel a bit out my depth so I would not mind someone else having a look before merging this.

File: tasks.py [hunk 1]
```diff
@@ -295,9 +295,9 @@ def ci_docs_build_pdf(c):
     c.run('make -f Makefile.sphinx -C docs latexpdf')
 
 @task
-def ci_docs_upload_master(c, docs_host, pdf, username, directory=""):
+def ci_docs_upload_master(c, docs_host, pdf, username, product, directory=""):
     c.run(f"rsync -crv --delete --no-p --chmod=g=rwX --exclude '*~' ./docs/_build/html-docs/ {username}@{docs_host}:{directory}")
-    c.run(f"rsync -crv --no-p --chmod=g=rwX --exclude '*~' ./docs/_build/html-docs.tar.bz2 {username}@{docs_host}:{directory}")
+    c.run(f"rsync -crv --no-p --chmod=g=rwX --exclude '*~' ./docs/_build/{product}-html-docs.tar.bz2 {username}@{docs_host}:{directory}")
```
Comment (Habbie) [inline tasks.py:300]:
```suggestion
    c.run(f"rsync -crv --no-p --chmod=g=rwX --exclude '*~' ./docs/_build/{product}-html-docs.tar.bz2 {username}@{docs_host}:{directory}/html-docs.tar.bz2")
```

**Review by Habbie — CHANGES_REQUESTED** [review]
one request, otherwise good!

**Review by Habbie — APPROVED** [review]

**Comment by fredmorcos** [comment]
Reminder to merge 😄

**Comment by rgacogne** [comment]
This appears to be failing: https://github.com/PowerDNS/pdns/actions/runs/4076304690/jobs/7023858077

**Comment by rgacogne** [comment]
```
Run inv ci-docs-upload-master --docs-host="${DOCS_HOST}" --pdf="PowerDNS-Authoritative.pdf" --username="docs_powerdns_com" product="auth" --directory="/${AUTH_DOCS_DIR}/"
Warning: Permanently added the ECDSA host key for IP address '188.166.104.92' to the list of known hosts.
rsync: change_dir "/home/runner/work/pdns/pdns//./docs/_build/html-docs" failed: No such file or directory (2)

rsync -crv --delete --no-p --chmod=g=rwX --exclude '*~' ./docs/_build/html-docs/ docs_powerdns_com@web1.powerdns.com://authoritative/
  sending incremental file list
  
  rsync error: some files/attrs were not transferred (see previous errors) (code 23) at main.c(1207) [sender=3.1.3]
  sent 32 bytes  received 12 bytes  17.60 bytes/sec
  total size is 0  speedup is 0.00
  Error: Process completed with exit code 23.
  ```

**Comment by fredmorcos** [comment]
> This appears to be failing: https://github.com/PowerDNS/pdns/actions/runs/4076304690/jobs/7023858077

Fix in #12503

**Pull request merged**